### PR TITLE
Combine node and npm publish workflows

### PR DESCRIPTION
**Pull request by donavanbecker**
### Changes
- one workflow for master branch
- one workflow for beta branch

### Go ahead and merge when ready.

### CODE REVIEW

**Comment by oznu** [comment]
This has removed the `on-PR` workflow?

File: .github/workflows/nodejs.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
+name: NodeJS
+
+on:
+  push:
+    branches: master
+  pull_request:
```
Comment (donavanbecker) [inline .github/workflows/nodejs.yml:6]: @oznu it is on the master branch.

File: .github/workflows/nodejs-beta.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
+# This workflow will run tests using node and then publish a package to GitHub Packages when a release is created
+# For more information see: https://help.github.com/actions/language-and-framework-guides/publishing-nodejs-packages
+
+name: Node-CI Beta
+
+on:
+  push:
+    branches:
+      - beta
```
Comment (donavanbecker) [inline .github/workflows/nodejs-beta.yml]: @oznu but not on beta

**Comment by donavanbecker** [comment]
> This has removed the `on-PR` workflow?

only on the beta branch. master branch still has it.

**Comment by oznu** [comment]
I think the current workflows, which are in beta right now, are fine as they are:

https://github.com/homebridge/homebridge/tree/beta/.github/workflows

**Pull request closed without merging**